### DISCUSSION TITLE
SPARQL query profiler?

### DISCUSSION THREAD

**Discussion by reckart** [eclipse-rdf4j/rdf4j#3873] [Q&A] [answered]
Does RDF4J have some kind of query profiler? Some kind of query execution report that would relate query time to a SPARQL query?

**Comment by hmottestad** [comment, marked as answer]
https://rdf4j.org/documentation/programming/repository/#explaining-queries

**Reply by naturzukunft** [reply]
> https://rdf4j.org/documentation/programming/repository/#explaining-queries

I was pleased to read this today and wanted to test it right away for my GraphQuery. Unfortunately it is an UnsupportedOperationException there.

**Reply by hmottestad** [reply]
Are you using a local or remote repository?

**Reply by naturzukunft** [reply]
In deed a remote one! in the local one everything works! 
Thanks!
